win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2291,9 +2289,104 @@ IntGdiAddFontResourceEx(PUNICODE_STRING FileName, DWORD Characteristics,
 }
 
 INT FASTCALL
-IntGdiAddFontResource(PUNICODE_STRING FileName, DWORD Characteristics)
+IntGdiAddFontResourceEx(
+    IN const UNICODE_STRING *FileName,
+    IN DWORD Characteristics,
+    IN DWORD dwFlags,
+    IN DWORD cFiles,
+    IN DWORD cwc)
 {
-    return IntGdiAddFontResourceEx(FileName, Characteristics, 0);
+    PWSTR pchFile = FileName->Buffer;
+    SIZE_T cchFile;
+    INT ret = 0;
+
+    while (cFiles--)
+    {
+        _SEH2_TRY
+        {
+            // Security issue: FileName should be terminated by double '\0'
```

Comment:
Are you really sure this is the purpose?

Reply:
Deleted comment and code.